# Descriptive Measures

In statistics, Descriptive measures are brief descriptive coefficients that summarize a given data set, which can be either a representation of the entire population or some sample of population. 

So, here we are going to describe about the following descriptive measures of statistics in Jupyter notebook using various python libraries.

- Measures of Central Tendency
- Measures of Variation

#  1) Measures of Central Tendency

Descriptive measures that indicate where the center or most typical value of a data set lies are called Measures of Central Tendency. These are often called averages. 

Here, We will discuss about the most important measures of central tendency which are the Mean, Meadian and Mode. 
The mean and median apply only to quantitative data, whereas the mode can be used with either quantitative or qualitative data. 

# Mean

The mean of a dataset is its arithmatic mean. The mean of dataset is the sum of the observations divided by the number of observations.

# Example

Here, we see one example...

We have 9 Asian Countries GDPs as follows : 7.5, 5.5, 10.6, 8.2, 14.2, 6.8, 9.2, 5.9, 4.2

Now we have to find average GDP of this Asian Countries then We will find it as below.


In [9]:
#Here we store all GDPs in one variable named asiangdp.

asiangdp = (7.5, 5.5, 10.6, 8.2, 14.2, 6.8, 9.2, 5.9, 4.2)

#We have to import NumPy library for doing some descriptive statistics.

import numpy as np

x = np.mean(asiangdp)
print(x)

#Press Shift+Enter to get the answer.

8.011111111111113


So, We got the Mean of 9 Asian Country's GDP. And it is 8.011  

# Median

Another frequently used measure of center is the median. Essentially, the median of a data set is the number that divides the bottom 50% of the data from the top 50%. The median of a data set is the middle value in its ordered list.

Arrange the data in increasing order.

- If the number of observations is odd, then the median is the observation exactly in the middle of the ordered list.
- If the number of observations is even, then the median is the mean of the two middle observations in the ordered list.

In both cases, if we let n denote the number of observations, then the median is at position (n + 1)/2 in the ordered list.

# Example

We have a record of 16 student's height in class as : 175, 185, 160, 163, 159, 180, 163, 174, 159, 181, 178, 169, 170, 174, 185, 167

Now we have to find the median height of the student in this particular class. So, We will code as following :


In [7]:
#Here We store all heights in one variable named studentsheights.

studentsheights = (175, 185, 160, 163, 159, 180, 163, 174, 159, 181, 178, 169, 170, 174, 185, 167)

#We have already imported NumPy library in previous code of Mean. So, we don't need to import it again ! 

y = np.median(studentsheights)
print(y)

#Press Shift+Enter to get the answer.

172.0


So We have the median height of the students in class is 172.

# Mode

The mode of a data set is its most frequently occurring value.

Find the frequency of each value in the data set.

- If no value occurs more than once, then the data set has no mode.
- Otherwise, any value that occurs with the greatest frequency is a mode of the data set.

# Example

Here, We have some records which contains the marks of 10 students as : 75, 89, 95, 45, 73, 66, 75, 59, 75, 81

Now, We have to find the mode of the student's marks. We will code as following:


In [10]:
#Here We store all marks in one variable named studentsmarks.

studentsmarks = (75, 89, 95, 45, 73, 66, 75, 59, 75, 81)

#Before, We imported NumPy Library but, to find Mode We have to import stats from SciPy Library. 

from scipy import stats

z = stats.mode(studentsmarks)
print(z)

#Press Shift+Enter to get the answer.

ModeResult(mode=array([75]), count=array([3]))


So, here we can see that the Mode is 75 which is repeated 3 times.


# 2) Measures of Variation

There are three measures of variability and they are:

- Range
- InterQuartile Range
- Standard Deviation 
- Variance

# Range

The range is simply the difference between the largest score (the maximum value) and the smallest score (the mininum value) of a distribution. This can give quick sense of the spread of the data.

# Example

Here, We have data of some speeds recorded for the cars as : 86, 65, 109, 26, 59, 69, 79, 90, 43, 120, 71, 63, 88, 104, 57

Let's see it on cell:

In [2]:
#Here We store all speed records in one variable named speed.

speed = [86, 65, 109, 26, 59, 69, 79, 90, 43, 120, 71, 63, 88, 104, 57]

#We define function for range as below:

def getrange(speed):
    return max(speed) - min(speed)

print(getrange(speed))

#Press Shift+Enter to get the answer.

94


Here, We get the range 94. e.g. 120 - 26 = 94 (max_value - min_value) 

# InterQuartile Range (IQR)

The IQR is the difference between the score that marks the 75th percentile which is a third quartile and the score that marks the 25th percentile which is a first quartile.

# Example  

Here, We taking the record of speed again which are as follows: 86, 65, 109, 26, 59, 69, 79, 90, 43, 120, 71, 63, 88, 104, 57

Let's check it out.

In [4]:
#Here We store all speed records in one variable named speed.

speed = [86, 65, 109, 26, 59, 69, 79, 90, 43, 120, 71, 63, 88, 104, 57]

#We will import iqr function from scipy.stats library

from scipy.stats import iqr
print(iqr(speed))

#Press Shift+Enter to get the answer.

28.0


Here we get 28 as the InterQuartile Range (IQR).

# Standard Deviation

The standard deviation indicates how far, on average, the observations in the sample are from the mean of the sample. So, if the average score for a distribution is 10 and any individual score is 12 then the deviation is 2. 

This is a very useful statistic because it provides a handly measure of how spread out the scores are in the distribution.

# Example

We have a records of the students' marks of one subject as : 45, 88, 71, 59, 82, 96, 62, 76, 85, 59.

We will find the standard deviation of students'marks.

In [8]:
#Here We store all students' marks in one variable named smarks.

smarks = [45, 88, 71, 59, 82, 96, 62, 76, 85, 59]

#Here we will import Statistics library.

import statistics
statistics.stdev(smarks)

#Press Shift+Enter to get the answer.

15.930752929127012

Sp, here we get 15.93 as the standard deviation. 

# Variance

The variance provides a statistical average of the amount of dispersion in a distribution of scores.

The square of the difference between an individual score in a distribution and the mean for the distribution is called squared deviation. And the sum of the squared deviations divided by the total number of records(data).

# Example

Here, We take the same data as we used in previous cell, the marks of students as : 45, 88, 71, 59, 82, 96, 62, 76, 85, 59

Let's see how we can find variance.

In [9]:
#Here We store all students' marks in one variable named smarks.

smarks = [45, 88, 71, 59, 82, 96, 62, 76, 85, 59]

#We will import statistics library.

import statistics
statistics.variance(smarks)

#Press Shift+Enter to get the answer.

253.7888888888889

Here, We get 253.8 as Variance which is the squared value of standard deviation which we found for same data in previous cell. e.g. (15.93^2).

# References

- Introductory Statistics by Neil A. Weiss
- Statistics in Plain English by Timothy C. Urdan
- Investopedia.com
- Geeksforgeeks.org
- Pluralsight.com
- W3Schools.com

# ------------------------------------------------------------------------------------------------------

Thanks for check it out !

This is my first jupyter notebook blog, so if there is any mistake then please give some needed instruction, so I can gain more knowledge. 

Any feedback would be appreciated. 
Thanks Again !